### ISSUE TITLE
ipywidget dropdown does not start with empty value - even when setting to None

### ISSUE THREAD

**Issue by acelere** [jupyter-widgets/ipywidgets#2086]
I am trying to build a dropdown that starts empty (by setting its "value=None". I want to force the user to select something.
I have found this [issue/fix](https://github.com/jupyter-widgets/ipywidgets/issues/612), where the dropdown value is set to "None"; However it only works after the widget is shown on screen and the user tries to select something.

I have ipywidgets 7.2.1 running and this is what it looks like (sorry for the bad gif quality)
<p></p>

![dd_bug2](https://user-images.githubusercontent.com/12059585/40432667-38c849ea-5e78-11e8-8e21-767d70634407.gif)
<p></p>
In my view, it should be possible to start with an empty dropdown.
<p></p>
Thanks

**Comment by jasongrout** [comment]
Agreed! I'm not sure if the issue is on the python side or the js side. Do you want to try to fix it? 

In the js side, it looks like the rendering code at https://github.com/jupyter-widgets/ipywidgets/blob/master/packages/controls/src/widget_selection.ts#L77 calls the same update function at https://github.com/jupyter-widgets/ipywidgets/blob/master/packages/controls/src/widget_selection.ts#L100, so it appears that the js is all right. It appears that maybe the index getting transmitted initially is not null.

The python code for setting the index initially given a value is at https://github.com/jupyter-widgets/ipywidgets/blob/766cad54a47c07520e9d695534c4664c3391e7ec/ipywidgets/widgets/widget_selection.py#L185.

**Comment by jasongrout** [comment]
Wait, what browser are you using? It works fine for me in FF and Chrome on macOS, but doesn't work in Safari.

**Comment by acelere** [comment]
Hummm. 

But note that I could only blank out the dropdown AFTER I try to select something. If I render the dd and immediately set the value to None, it does not blank out. <p></p> 
 So I guess something is going on the sync side for values set to None.
<p></p>

Yes, as a workaround, explicit '' would work.
However, I don't like it because then the callbacks need to have a case for the empty value if the user selects it.

**Comment by jasongrout** [comment]
When I was exploring this issue before, when making it possible to set the value to None, I realized that different browsers had different behavior regarding allowing a dropdown to be set to a null selection. It may be that we just conclude that Edge and Safari don't allow it.

If you'd like to continue investigating, I'd suggest some experiments modifying the js code to try to set the index appropriately on creation for these browsers, and see if it works. To me, it looks like it won't work in Safari at all. It may work in Edge.

**Comment by acelere** [comment]
OK. give me some days to try it out.
Thanks for following up!
